Source/Core/Core/NetworkCaptureLogger.cpp

```diff
@@ -0,0 +1,222 @@
+// Copyright 2020 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Core/NetworkCaptureLogger.h"
+
+#include <array>
+#include <cerrno>
+#include <chrono>
+#include <cstring>
+#include <iomanip>
+#include <iterator>
+#include <sstream>
+
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Common/Network.h"
+#include "Common/PcapFile.h"
+#include "Core/Config/MainSettings.h"
+#include "Core/ConfigManager.h"
+
+namespace Core
+{
+NetworkCaptureLogger::~NetworkCaptureLogger() = default;
+
+void DummyNetworkCaptureLogger::LogSSLRead(const void* data, std::size_t length, s32 socket)
+{
+}
+
+void DummyNetworkCaptureLogger::LogSSLWrite(const void* data, std::size_t length, s32 socket)
+{
+}
+
+void DummyNetworkCaptureLogger::LogRead(const void* data, std::size_t length, s32 socket,
+                                        sockaddr* from)
+{
+}
+
+void DummyNetworkCaptureLogger::LogWrite(const void* data, std::size_t length, s32 socket,
+                                         sockaddr* to)
+{
+}
+
+NetworkCaptureType DummyNetworkCaptureLogger::GetCaptureType() const
+{
+  return NetworkCaptureType::None;
+}
+
+void BinarySSLCaptureLogger::LogSSLRead(const void* data, std::size_t length, s32 socket)
```

Comment:
Not a big deal but I wonder if we could combine this and writing into one function